spopt/locate/coverage.py

```diff
@@ -208,22 +208,30 @@ def from_cost_matrix(
             )
 
         if demand_quantity_arr is not None:
+
+            sum_demand = demand_quantity_arr.sum()
+            sum_capacity = facility_capacity_arr.sum()
+            if sum_demand > sum_capacity:
+                raise ValueError(
+                    f"Infeasible model. Demand greater than capacity "
+                    f"({sum_demand} > {sum_capacity})."
+                )
+
             FacilityModelBuilder.add_client_assign_integer_variable(
                 lscp, r_cli, r_fac, "z[{i}_{j}]", lp_category=pulp.LpContinuous
             )
 
             FacilityModelBuilder.add_facility_capacity_constraint(
                 lscp,
                 lscp.problem,
-                lscp.aij,
                 facility_capacity_arr,
                 demand_quantity_arr,
                 r_fac,
                 r_cli,
             )
 
             FacilityModelBuilder.add_client_demand_satisfaction_constraint(
-                lscp, lscp.problem, r_cli, r_fac
+                lscp, lscp.problem, lscp.aij, r_cli, r_fac
```

Comment:
is it redundant to pass in lscp, lscp.problem, lscp.aij as separate args, when the latter two are attributes/members of the former?

Reply:
This is a very good point. I'd say let's open an issue to standard are streamline variable & constraint creation. Then we can merge this PR in and work on that after?

Reply:
xref #317

Reply:
So I'll go ahead with the merge and get to work on #317.